data/routers/userInfoRouters.js

```diff
@@ -0,0 +1,72 @@
+const express = require("express");
+const bcrypt = require("bcryptjs");
+const router = express.Router();
+const db = require("../models/userInfoModel");
+const session = require("express-session");
+
+router.use(
+  session({
+    name: "notsession",
+    secret: "not a gnelf not a gnoblin",
+    cookie: {
+      maxAge: 1 * 24 * 60 * 60 * 1000,
+      secure: false
+    },
+    httpOnly: true,
+    resave: false,
+    saveUninitialized: false
+  })
+);
+
+router.get("/", (req, res) => {
+  db.get().then(userInfo => {
+    res.json(userInfo);
+  });
+});
+
+router.get("/users", (req, res) => {
+  db.getAllUsernames()
+    .then(usernameList => {
+      if (req.session && req.session.userId) {
+       const usernameArray = [];
+        usernameList.forEach(user => {
+          usernameArray.push(user.username);
+        });
+        res.json({usernames: usernameArray });
```

Comment:
Don't request the users from the database until you know that the user is authenticated. This would be a good opportunity to use middleware to check if the user is authenticated.